pkg/cmd/gist/edit/edit.go

```diff
@@ -222,3 +297,36 @@ func updateGist(apiClient *api.Client, hostname string, gist *shared.Gist) error
 
 	return nil
 }
+
+func processFiles(filename string) (string, bool, error) {
+	fileExists := false
+
+	if fi, err := os.Stat(filename); err != nil {
+	} else {
+		switch mode := fi.Mode(); {
+		case mode.IsDir():
+			return "", false, fmt.Errorf("found directory %s", filename)
+		case mode.IsRegular():
+			fileExists = true
+		}
+	}
+
```

Comment:
why continue execution, here? In my testing `os.Stat` was sufficient for finding the various file paths i threw at this.